### ISSUE TITLE
[SafeArea][WinUI] SafeArea should be using XamlRoot and not Window.Current

### ISSUE THREAD

**Issue by kazo0** [unoplatform/uno.toolkit.ui#382]
In spots like [here](https://github.com/unoplatform/uno.toolkit.ui/blob/0d2a3606cd1398e3bf78589ac59229f39adb618a/src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs#L195) and multiple other places, we should be using the XamlRoot [Uno](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/XamlRoot.cs) [Windows](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.xamlroot?view=winrt-22621) since Window.Current returns null on WinUI.

**Comment by agneszitte** [comment]
@MartinZikmund is this something to consider for multi-window support ?

**Comment by MartinZikmund** [comment]
yes, very likely, on WinAppSDK and Uno.WinUI apps `Window.Current` would not work
